src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -238,32 +239,37 @@ class LicenceService(
 
     val username = SecurityContextHolder.getContext().authentication.name
 
-    val newConditions = licenceEntity.additionalConditions.toMutableList()
-
-    newConditions.add(
-      AdditionalCondition(
-        conditionVersion = licenceEntity.version!!,
-        conditionType = request.conditionType,
-        conditionCode = request.conditionCode,
-        conditionText = request.conditionText,
-        expandedConditionText = request.expandedText,
-        conditionCategory = request.conditionCategory,
-        licence = licenceEntity,
-        conditionSequence = request.sequence
-      )
+    val existingAdditionalConditions = licenceEntity.additionalConditions.toMutableList()
+    var newAdditionalCondition = AdditionalCondition(
+      conditionVersion = licenceEntity.version!!,
+      conditionType = request.conditionType,
+      conditionCode = request.conditionCode,
+      conditionText = request.conditionText,
+      expandedConditionText = request.expandedText,
+      conditionCategory = request.conditionCategory,
+      licence = licenceEntity,
+      conditionSequence = request.sequence
+    )
+    if (isAdditionalConditionOf14B(request.conditionCode) && hasCorrectStatusCodeForNew14B(licenceEntity)) {
+      newAdditionalCondition = additionalConditionRepository.saveAndFlush(newAdditionalCondition)
```

Comment:
is this needed? There's cascade all on `licence -> additional condition -> additional condition data`